components/brave_extension/extension/brave_extension/state/shieldsPanelState.ts

```diff
@@ -129,8 +129,9 @@ export const updateShieldsIconBadgeText: shieldState.UpdateShieldsIconBadgeText
   const tab: shieldState.Tab = state.tabs[tabId]
   if (tab) {
     const total = getTotalResourcesBlocked(tab)
+    const text: string = total > 99 ? '99+' : total > 0 ? total.toString() : ''
     // do not show any badge if there are no blocked items
-    setBadgeText(tabId, total > 99 ? '99+' : total > 0 ? total.toString() : '')
+    setBadgeText(tabId, text, state.persistentData.statsBadgeVisible)
```

Comment:
Why pass in this boolean to a generic API and not just only pass text if statsBadgeVisible is true?

Reply:
we need to pass empty strings to override the previous call. otherwise we will need to update the page to have the change visible

Reply:
Suggestion is to pass empty string and keep the decision in the caller layer and keep `setBadgeText` a simple API. `setBadgeText(tabId, statsBadgeVisible ? '' : text)`